### PR TITLE
Export rdflib NamedNodes

### PR DESCRIPTION
**Pull request by JordanShurmer**
This is for #2. See the issue for more discussion.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,20 @@
-# solid-namespace
+# Solid Namespace
 [![](https://img.shields.io/badge/project-Solid-7C4DFF.svg?style=flat)](https://github.com/solid/solid)
 [![NPM Version](https://img.shields.io/npm/v/solid-namespace.svg?style=flat)](https://npm.im/solid-namespace)
 
 A collection of common RDF namespaces used in the Solid project.
 
+solid-namespace is meant to be used with [rdflib.js](/linkeddata/rdflib.js). It exports a collection of `NamedNode` values for commonly used namespaces, thereby shortcutting the need to call `$rdf.namespace(...)`.
```
Comment (RubenVerborgh) [inline README.md]: Meant to be used with any [RDF/JS](http://rdf.js.org/)-compatible library, including rdflib.js.

File: package.json [hunk 2]
```diff
@@ -22,15 +22,21 @@
...
   "bugs": {
     "url": "https://github.com/solid/solid-namespace/issues"
   },
   "homepage": "https://github.com/solid/solid-namespace",
   "dependencies": {
-    "rdf-ns": "^0.1.0"
+    "rdf": "^4.0.1"
+  },
+  "devDependencies": {
+    "standard": "^12.0.1"
```
Comment (RubenVerborgh) [inline package.json]: Out of scope for this PR?

Comment (JordanShurmer) [reply package.json]: true. removing.

**Comment by JordanShurmer** [comment]
@RubenVerborgh I've reworked my fork to work as described above and updated the README and JS doc comments etc.

Let me know what you think!

File: README.md [hunk 3]
```diff
@@ -1,14 +1,37 @@
...
 
+solid-namespace can be used with any RDF/JS-compatible library (e.g. [rdflib.js](/linkeddata/rdflib.js)). 
+
+
 ## Usage
 
+You can use this library in two ways. 
+
+1. With a RDF JS library to get NamedNodes
+2. Without a library to get url strings
```
Comment (RubenVerborgh) [inline README.md]: Problem is that 1. now works with a function, but 2. as an object. My solution makes both a function.

File: README.md [hunk 4]
```diff
@@ -1,14 +1,37 @@
...
+
+### Without a rdf library
+
 ```js
-var rdf = require('rdflib')  // optional
-var vocab = require('solid-namespace')(rdf)  // or require('solid-namespace')()
-console.log(vocab.foaf('name'))  
-// -> NamedNode(<http://xmlns.com/foaf/0.1/name>)
+const ns = require('solid-namespace')();
+console.log(ns.foaf('name')); // -> "http://xmlns.com/foaf/0.1/name"
```
Comment (RubenVerborgh) [inline README.md]: That won't work.

**Comment by JordanShurmer** [comment]
Interesting.. didn't realize that Namespace wasn't part of RDF/JS! `$rdf.Namespace` is what the inrupt/solid guides have you use, so I assumed that was the standard thing, and that `namedNode` was the rdflib.js specific thing. Got it directly backwards 🤦‍♂️ 

I totally botched the exporting of an object when no rdflib is given too. 

I'll get this eventually! 😑

**Comment by RubenVerborgh** [comment]
> `$rdf.Namespace` is what the inrupt/solid guides have you use

They might be a bit foo focused on rdflib (from my perspective, opinions vary).

> I'll get this eventually! 😑

Hey, thanks for persisting :-) The result is going to be something better than either of us would've done alone.

**Comment by JordanShurmer** [comment]
> The result is going to be something better than either of us would've done alone.

Definitely! 

Have a look now.. I also added tests to avoid my botching something obvious again :D

**Comment by JordanShurmer** [comment]
So how do we get this PR actually merged? Am I supposed to do something at this point?

**Comment by RubenVerborgh** [comment]
I don't have publish permissions on solid-namepaces, so I can merge but not publish.

Next time, could you please not bump the version? Because package maintainers usually have their own process for doing that.

**Comment by JordanShurmer** [comment]
> Next time, could you please not bump the version? Because package maintainers usually have their own process for doing that.

👍 understood, that makes total sense.

**Pull request merged**